optaplanner-examples/src/main/java/org/optaplanner/examples/app/ScoreDirectorFactoryBenchmarkApp.java

```diff
@@ -0,0 +1,219 @@
+import org.optaplanner.examples.flightcrewscheduling.domain.FlightCrewSolution;
+import org.optaplanner.examples.flightcrewscheduling.optional.score.FlightCrewSchedulingConstraintProvider;
+import org.optaplanner.examples.flightcrewscheduling.persistence.FlightCrewSchedulingXlsxFileIO;
+import org.optaplanner.examples.machinereassignment.domain.MachineReassignment;
+import org.optaplanner.examples.machinereassignment.domain.MrProcessAssignment;
+import org.optaplanner.examples.machinereassignment.persistence.MachineReassignmentFileIO;
+import org.optaplanner.examples.machinereassignment.solver.score.MachineReassignmentConstraintProvider;
+import org.optaplanner.examples.machinereassignment.solver.score.MachineReassignmentIncrementalScoreCalculator;
+import org.optaplanner.examples.nqueens.domain.NQueens;
+import org.optaplanner.examples.nqueens.domain.Queen;
+import org.optaplanner.examples.nqueens.solver.score.NQueensAdvancedIncrementalScoreCalculator;
+import org.optaplanner.examples.nqueens.solver.score.NQueensConstraintProvider;
+import org.optaplanner.examples.nqueens.solver.score.NQueensMapBasedEasyScoreCalculator;
+import org.optaplanner.examples.rocktour.domain.RockShow;
+import org.optaplanner.examples.rocktour.domain.RockStandstill;
+import org.optaplanner.examples.rocktour.domain.RockTourSolution;
+import org.optaplanner.examples.rocktour.optional.score.RockTourConstraintProvider;
+import org.optaplanner.examples.rocktour.persistence.RockTourXlsxFileIO;
+import org.optaplanner.examples.taskassigning.domain.Task;
+import org.optaplanner.examples.taskassigning.domain.TaskAssigningSolution;
+import org.optaplanner.examples.taskassigning.domain.TaskOrEmployee;
+import org.optaplanner.examples.taskassigning.solver.score.TaskAssigningConstraintProvider;
+import org.optaplanner.examples.vehiclerouting.domain.Customer;
+import org.optaplanner.examples.vehiclerouting.domain.Standstill;
+import org.optaplanner.examples.vehiclerouting.domain.VehicleRoutingSolution;
+import org.optaplanner.examples.vehiclerouting.domain.timewindowed.TimeWindowedCustomer;
+import org.optaplanner.examples.vehiclerouting.optional.score.VehicleRoutingConstraintProvider;
+import org.optaplanner.examples.vehiclerouting.persistence.VehicleRoutingFileIO;
+import org.optaplanner.examples.vehiclerouting.solver.score.VehicleRoutingEasyScoreCalculator;
+import org.optaplanner.examples.vehiclerouting.solver.score.VehicleRoutingIncrementalScoreCalculator;
+import org.optaplanner.persistence.common.api.domain.solution.SolutionFileIO;
+
+// TODO include cheaptime
+// TODO include coachshuttlegathering
+// TODO include dinnerparty
+// TODO include examination
+// TODO include investment
+// TODO include meetingscheduling
+// TODO include nurserostering
+// TODO include pas
+// TODO include projectjobscheduling
+// TODO include scrabble
+// TODO include tennis
+// TODO include travelingtournament
+// TODO include tsp
+public class ScoreDirectorFactoryBenchmarkApp {
```

Comment:
Normally, we do scoreDirector benchmarks simply by adding a bencharkConfig.xml and a line in the BenchmarkApp.

Is there a reason we can't cover these benchmarks by adding
optaplanner-examples/src/main/resources/org/optaplanner/examples/app/benchmark/generalOptaPlannerScoreDirectorBenchmarkConfigTemplate.xml.ftl and a line in GeneralOptaPlannerBenchmarkApp?

Reply:
Ah, I see, the easy score directors are different every time. Same for inc's etc.

Reply:
We might want to take this a step further, if we go down this route. See 
CloudBalancingSolveAllTurtleTest#overwritingEasyScoreCalculatorClass

If we have a register class for all score director variants, we can automate more testing and benchmarking.

Reply:
@ge0ffrey Not sure how this helps with benchmarking. I simply need something to run a benchmark of all SDF over all defined problems.

But perhaps I'm just not getting you. Would you mind explaining yourself a bit more?